apps/typegpu-docs/src/content/examples/simulation/fluid-with-atomics/index.ts

```diff
@@ -264,17 +264,15 @@ const vertex = tgpu['~unstable'].vertexFn({
 })((input) => {
   const w = size.$.x;
   const h = size.$.y;
-  const x = (((d.f32(input.idx % w) + input.squareData.x) / d.f32(w) - 0.5) *
-    2 *
-    d.f32(w)) /
-    d.f32(std.max(w, h));
-  const y = ((d.f32((input.idx - (input.idx % w)) / d.f32(w)) +
-        d.f32(input.squareData.y)) /
-      d.f32(h) -
-    0.5) *
-    2 *
-    d.f32(h) /
-    d.f32(std.max(w, h));
+  const gridX = input.idx % w;
+  const gridY = (input.idx - gridX) / w;
+  const maxDim = d.f32(std.max(w, h));
+
+  const x = ((d.f32(gridX) + input.squareData.x) / d.f32(w) - 0.5) * 2 *
+    d.f32(w) / maxDim;
+  const y = ((d.f32(gridY) + input.squareData.y) / d.f32(h) - 0.5) * 2 *
+    d.f32(h) / maxDim;
```

Comment:
We could simplify it to this
```suggestion
  const gridX = input.idx % w;
  const gridY = d.u32(input.idx / w);
  const maxDim = std.max(w, h);

  const x = (2 * (d.f32(gridX) + input.squareData.x) - d.f32(w)) / maxDim;
  const y = (2 * (d.f32(gridY) + input.squareData.y) - d.f32(h)) / maxDim;
```